docs/source/models.rst

```diff
@@ -147,6 +150,72 @@ class has four key functions:
 * :py:func:`load <mlflow.models.Model.load>` to load a model from a local directory or
   from an artifact in a previous run.
 
+
+Models From Code
+^^^^^^^^^^^^^^^^
+
+.. note::
+    The Models from Code feature is available in MLflow versions 2.12.2 and later. This feature is experimental and may change in future releases.
+
+The Models from Code feature allows you to define and log models directly from Python code. This feature is particularly useful when you want to 
+log models that can be effectively stored as a code representation (models that do not need optimized weights through training) or applications 
+that rely on external services (e.g., LangChain chains). 
+
+.. note::
+    This feature is only supported for **LangChain** and **PythonModel** models.
+
+In order to log a model from code, you can leverage the :py:func:`mlflow.models.set_model` API. This API allows you to define a model by specifying
+an instance of the model class directly within the file where the model is defined. When logging such a model, a
+file path is specified (instead of an object) that points to the Python file containing both the model class definition and the usage of the 
+``set_model`` API applied on an instance of your custom model. 
+
+The figure below provides a comparison of the standard model logging process and the Models from Code feature for models that are eligible to be 
+saved using the Models from Code feature:
+
+.. figure:: _static/images/models/models_from_code.png
+    :alt: Models from Code
+    :width: 60%
+    :align: center
+
+For example, defining a model in a separate file named ``my_model.py``:
+
+.. code-block:: python
+
+    import mlflow
+    from mlflow.models import set_model
```

Comment:
Model-from-code feature doesn't support importing custom module outside the `.py` file, right? Considering the number of questions we've got about pyfunc `code_paths`, it may be helpful to explicitly mention that limitation.

Reply:
We can make it supporting `code_paths`, (haven't implemented this yet)

the "model code" is saved as a separate code file so it is orthogonal to code_paths feature

Reply:
great callout. I'll add a note
@WeichenXu123 good idea - can we file a backlog (next sprint) JIRA to enhance the automated code_paths feature to support external reference imports for models from code?